Library/Homebrew/cmd/leaves.rb

```diff
@@ -15,14 +15,31 @@ def leaves_args
       description <<~EOS
         List installed formulae that are not dependencies of another installed formula.
       EOS
+      switch "-r", "--installed-on-request",
+             description: "Only list leaves that were manually installed."
+      switch "-p", "--installed-as-dependency",
+             description: "Only list leaves that were installed as dependencies."
 
       named_args :none
     end
   end
 
+  def installed_on_request?(leaf)
+    Tab.for_keg(leaf.any_installed_keg).installed_on_request
+  end
+
+  def installed_as_dependency?(leaf)
+    Tab.for_keg(leaf.any_installed_keg).installed_as_dependency
+  end
```

Comment:
Where are these methods used? If you're trying to use them below: I think you need 
```ruby
leaves_list.select!(&method(:installed_on_request?))
```

Reply:
They're defined in the leaves command file. Since they aren't methods of `Formula`, `method` has to be used. Thanks!